Sources/GoogleCloudProvider/Provider.swift

```diff
@@ -8,12 +8,12 @@
 import Vapor
 
 public struct GoogleCloudProviderConfig: Service {
-    public let project: String
-    public let privateKey: String
-    
-    public init(projectId: String, rsaPrivateKey: String) {
-        project = projectId
-        privateKey = rsaPrivateKey
+    let project: String
+    let serviceAccountCredentialPath: String?
+
+    public init(project: String, credentialFile: String? = nil) {
```

Comment:
@bmhatfield We also don't need a project name anymore since it'll be in the ENV vars or the credential file.

Reply:
Good point about it being in the service-account credentials file. 

However, the only environment variable it's in currently in this codebase is test related. For people using the application_default_credentials, that implies that it will be unconfigured.

While I personally don't mind using environment variables to configure everything, I thought it was best to keep a configuration value for people who want it.

Thoughts?

Reply:
As an additional datapoint, it's also possible for a service account or application default credentials to be valid in multiple Google Cloud Projects. I think that this means we can't trust the value in the service account JSON in all cases, as a user might want to use a different project.

Given that, I recommend we keep this configuration value.

Reply:
@bmhatfield 
Sure I can agree with that. At the very least we can make it optional maybe? because for my use case specifically I know that I won't be deploying to anything but GCP and all my $PROJECT_ID etc will be on the machines I deploy to so I wouldn't have a need to even bother configuring it. Same for my CI/CD pipeline which is built with Container Builder.

Reply:
Sure! I'll add functionality to make it pull from `PROJECT_ID`, the provider config, and the config file (in that order).

Reply:
Awesome 👍 And just remember to update the README with the new config instructions.

Reply:
Done to both!